### PR TITLE
Get rid of reflect in mempool code

### LINKED ISSUE #38: Get rid of reflection in mempool
To eliminate [use of reflect in mempool integration](https://github.com/lazyledger/optimint/pull/34/files#diff-375d57e386f20eaa5f09f02bb9d28bfc48ac3dca18d0325f59492208219e5618R117), we need to make changes into mempool code (change visibility of a struct).

As described in #37, we decided to copy deps into optimint repository.

### PR DESCRIPTION
**Pull request by tzdybal**
This PR copies the mempool dependency into code tree and integrate the changes, to avoid using `reflect` in mempool related code.

Depends on #34.
Relates to #37.
Resolves #38.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -113,10 +112,8 @@ func (n *Node) mempoolPublishLoop(ctx context.Context) {
 		// send transactions
 		for {
 			n.Logger.Debug("Gossiping...")
-			// TODO(tzdybal): refactor lazyledger-core to avoid reflection
-			v := reflect.Indirect(reflect.ValueOf(next.Value))
-			f := v.FieldByName("tx")
-			tx := f.Bytes()
+			memTx := next.Value.(*mempool.MempoolTx)
```
Comment (liamsi) [inline node/node.go:115]: What happens if the cast fails?

Comment (tzdybal) [reply node/node.go:115]: We will see a panic in tests and will be able to fix the issue.
Initially I created idiomatic error handling for this cast... But in this particular case I think we should just panic.

Comment (liamsi) [reply node/node.go:115]: I see, I briefly thought this was the other way around: you could send a Tx and crash the process / node but in this case the process is in control.

File: mempool/clist_mempool.go [hunk 2]
```diff
@@ -0,0 +1,780 @@
+package mempool
```
Comment (liamsi) [inline mempool/clist_mempool.go:1]: Maybe add do.go file documenting where this is coming from, why it was copied and what was modified.

File: mempool/clist_mempool.go [hunk 3]
```diff
@@ -0,0 +1,780 @@
...
+	_, err := mem.proxyAppConn.FlushAsync(ctx)
+	if err != nil {
+		mem.logger.Error("Can't flush txs", "err", err)
+	}
+}
+
+//--------------------------------------------------------------------------------
+
+// MempoolTx is a transaction that successfully ran
+type MempoolTx struct {
```
Comment (liamsi) [inline mempool/clist_mempool.go:672]: That is now exported should be the only change from the tendermint mempool, right?

File: mempool/mempool.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+// transaction if false is returned. An example would be to ensure a
+// transaction doesn't require more gas than available for the block.
+type PostCheckFunc func(types.Tx, *abci.ResponseCheckTx) error
+
+// TxInfo are parameters that get passed when attempting to add a tx to the
+// mempool.
+type TxInfo struct {
+	// SenderID is the internal peer ID used in the mempool to identify the
+	// sender, storing 2 bytes with each tx instead of 20 bytes for the p2p.ID.
+	SenderID uint16
```
Comment (liamsi) [inline mempool/mempool.go:103]: Ah OK, this also makes this: https://github.com/lazyledger/optimint/pull/34#discussion_r612681302 obsolete, right?

**Review by liamsi — APPROVED** [review]
It would be helpful for reviewers to highlight which files were and how they were modified in github comments.

File: mempool/mempool.go (hunk 4 above)
Comment (tzdybal) [reply mempool/mempool.go:103]: Exactly

File: mempool/clist_mempool.go (hunk 3 above)
Comment (tzdybal) [reply mempool/clist_mempool.go:672]: Yes, this is the only change. I won't squash during a merge this time, to make this change more visible.

File: mempool/clist_mempool.go (hunk 2 above)
Comment (tzdybal) [reply mempool/clist_mempool.go:1]: Resolved in 6d91a8c.

File: mempool/clist_mempool.go (hunk 3 above)
Comment (liamsi) [reply mempool/clist_mempool.go:672]: Makes sense! Thanks.

**Pull request merged**